src/base/Solution.cpp

```diff
@@ -8,14 +8,39 @@
 
 #include "cantera/base/Solution.h"
 #include "cantera/thermo/ThermoPhase.h"
+#include "cantera/thermo/ThermoFactory.h"
 #include "cantera/kinetics/Kinetics.h"
+#include "cantera/kinetics/KineticsFactory.h"
 #include "cantera/transport/TransportBase.h"
+#include "cantera/transport/TransportFactory.h"
 
 namespace Cantera
 {
 
 Solution::Solution() {}
 
+  Solution::Solution(const std::string& infile, std::string name, std::string transport,
+                     std::vector<shared_ptr<Solution> > adjacent) {
+
+    // thermo phase
+    m_thermo = shared_ptr<ThermoPhase>(newPhase(infile, name));
+
+    // kinetics
+    std::vector<ThermoPhase*> phases;
+    for (auto & adj : adjacent) {
+        phases.push_back(adj->thermo().get());
+    }
+    phases.push_back(m_thermo.get());
```

Comment:
Shouldn't `m_thermo` be first in the `phases` list, per the documentation of `newKinetics`?

Reply:
Well, that means that the old class class wasn't consistent with the docs, as I kept things implemented like before. This will likely change some output_blessed files but not much else.

Reply:
Tests passed even after changing the order to the one consistent with `newKinetics`.

This is what I had based things on (which is superseded by `newKinetics`): https://github.com/Cantera/cantera/blob/07e0e2fb2ed67c47a16ba11c7c435a87ea4094eb/src/kinetics/importKinetics.cpp#L155-L160